swap_meet/vendor.py

```diff
@@ -1,2 +1,70 @@
+#wave 1
 class Vendor:
-    pass
\ No newline at end of file
+
+    def __init__(self, inventory=None):
+        self.inventory = inventory or []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+    
+#wave 2  
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+
+# wave 3
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+
+        self.remove(my_item)
+        other_vendor.add(my_item)
+
+        other_vendor.remove(their_item)
+        self.add(their_item)
+
+        return True
+    
+#wave 4
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+
+        my_first = self.inventory[0]
+        their_first = other_vendor.inventory[0]
+        return self.swap_items(other_vendor, my_first, their_first)
+    
+# Wave 6
+
+    def get_by_category(self, category):
+        result_Objects = []
```

Comment:
This variable breaks naming conventions for Python, local variable should be all lowercased:
```suggestion
        result_objects = []
```

I also suggest renaming this variable to better reflect what it holds. `result_Objects` doesn't give other developers information about what the result represents in our context. `objects_for_category` or `inventory_for_category` are some examples of names that tell readers more about the code.